src/lib_db.ml

```diff
@@ -20,15 +20,22 @@ type t =
   ; by_scope_name            : (string, scope) Hashtbl.t
   }
 
-let rec internal_name_scope t ~dir =
-  match Hashtbl.find t.by_internal_name dir with
-  | Some scope -> scope
-  | None ->
-    (* [create] ensures that [Hashtbl.find t.by_internal_name Path.root] is [Some _] so
-       this [Path.parent dir] is never called with [Path.root] *)
-    let scope = internal_name_scope t ~dir:(Path.parent dir) in
-    Hashtbl.add t.by_internal_name ~key:dir ~data:scope;
-    scope
+let internal_name_scope t ~dir =
+  let rec loop d =
+    match Hashtbl.find t.by_internal_name d with
+    | Some scope -> scope
+    | None ->
+      if Path.is_root d || not (Path.is_local d) then (
+        code_errorf "scope dir %s must be a descendant of %s"
+          (Path.to_string_maybe_quoted dir)
+          (Path.to_string_maybe_quoted t.anonymous_root)
+      );
+      (* [create] ensures that [Hashtbl.find t.by_internal_name Path.root] is
+         [Some _] so this [Path.parent dir] is never called with [Path.root] *)
```

Comment:
I just realised there is an inconsistency between this comment and the test. The comment is no longer true, right?